tools/nimgrep.nim

```diff
@@ -50,44 +58,132 @@ Options:
                -b:N   print N lines of leading context before every match
   --context:N, -c:N   print N lines of leading context before every match and
                       N lines of trailing context after it
+  --sortTime          order files by the last modification time -
+       -s[:asc|desc]  - ascending (default: recent files go last) or descending
   --group, -g         group matches by file
   --newLine, -l       display every matching line starting from a new line
+  --limit[:N], -m[:N] limit max width of lines from files by N characters (80)
+  --fit               calculate --limit from terminal width for every line
+  --onlyAscii, -o     use only printable ASCII Latin characters 0x20-0x7E
+                      (substitutions: 0 -> @, 1-0x1F -> A-_, 0x7F-0xFF -> !)
```

Comment:
somewhat serious bug: `$ nimgrep x bin/nim` hangs

Reply:
I cannot reproduce :-/

What's your OS? Could you compile with --gc:markandsweep (I don't quite trust my recent change to --gc:orc)?

Reply:
repro:
I'm on OSX, using iterm
```
XDG_CONFIG_HOME= $nim_140_X c tools/nimgrep.nim # (XDG_CONFIG_HOME to isolate from my environment)
tools/nimgrep x $nim_140_X
```

adding `--gc:markandsweep` or `--gc:arc` makes no difference

* this hangs: `tools/nimgrep x $nim_140_X`
* this does not hang: `tools/nimgrep x $nim_140_X | wc -l`
* this does not hang: `tools/nimgrep --onlyAscii $nim_140_X | wc -l`

this suggests that the culprit is binary data being printed to the terminal are being interpreted by the terminal, causing arbitrary behavior including hanging.

This should not be the default behavior (printing binary data to terminal can have potentially dangerous / malicious consequences), so I suggest making `--onlyAscii:on` the default, and users that know what they're doing can us explicit `--onlyAscii:off` to disable

We can bikeshed the `--onlyAscii` name separately